src/osd/ECTransaction.h

```diff
@@ -120,6 +132,28 @@ class ECTransaction : public PGBackend::PGTransaction {
     assert(len == bl.length());
     ops.push_back(AppendOp(hoid, off, bl, fadvise_flags));
   }
+  void write(
+    const hobject_t &hoid,
+    uint64_t off,
+    uint64_t len,
+    bufferlist &bl,
+    uint32_t fadvise_flags) {
+    if (len == 0) {
+        touch(hoid);
+        return;
+    }
+    offset_write = true;
```

Comment:
ECBackend use this to know whether the op is overwrite.
it looks awkward now, next I will refactor it along with AppendOp.